ndonnx/_core/_coreimpl.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) QuantCo 2023-2024
+# SPDX-License-Identifier: BSD-3-Clause
+
+from spox import Tensor, argument
+
+import ndonnx as ndx
+import ndonnx._data_types as dtypes
+import ndonnx.additional as nda
+from ndonnx._corearray import _CoreArray
+
+from ._interface import OperationsBlock
+from ._utils import validate_core
+
+
+class CoreOperationsImpl(OperationsBlock):
+    def make_array(self, shape, dtype, eager_value=None):
```

Comment:
I think there would be a cyclic import somewhere. I will refactor the module structure to accommodate this in a future PR.

Reply:
Using `TYPE_CHECKING` as done [here](https://github.com/Quantco/ndonnx/blob/84b8f4d08b7579b81a67393fc01c485d339abe35/ndonnx/_utility.py#L19) should provide a simple solution for the circular import.

Reply:
I'm aware of `TYPE_CHECKING` but unfortunately it is not sufficient. I will also need to annotate it as a "string" which I wanted to avoid until we have a better solution:

```python
if TYPE_CHECKING:
    from ndonnx._array import Array
    from ndonnx._data_types import CoreType, StructType

    Dtype = Union[CoreType, StructType]


class CoreOperationsImpl(OperationsBlock):
    def make_array(
        self,
        shape: tuple[int | None | str, ...],
        dtype: "Dtype",
        eager_value: np.ndarray | None = None,
    ) -> "Array":
```

I'm fine with doing this for now but perhaps we can aggregate the types somewhere in a subsequent PR.

Reply:
This should be fixed by doing
```python
from __future__ import annotations
```
as the very first import.

Reply:
Nice, thanks for the tip!